ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,104 @@
+# 39. Workspace Deprecation
+
+Some of the text was taken from - https://github.com/konflux-ci/architecture/pull/187
+
+## Status
+
+WiP
+
+## Context
+
+The purpose of this ADR is to revisit the workspace concept, understand the purpose
+of it, and offer an alternative implementation based on native Kubernetes APIs
+and other successful cloud native open source projects.
+
+We inherited the "workspace" concept from our early days when we expected to be integrated with the [KCP] project. Long ago (over a year at the time of this writing), we decided to remove [KCP] from the architecture of the Konflux project.
+
+In order to support multi-cluster deployment, [kubesaw] was used instead of KCP.
```

Comment:
None of these sentences is true.

> We inherited the "workspace" concept from our early days when we expected to be integrated with the [KCP] project. Long ago (over a year at the time of this writing), we decided to remove [KCP] from the architecture of the Konflux project.

As I mentioned in the other comment, the "workspace" API doesn't have anything in common with the "workspace concept" in kcp.

> In order to support multi-cluster deployment, [kubesaw] was used instead of KCP.

KubeSaw was used as the proven solution for user & namespace management. Multi-cluster was one of the benefits that KubeSaw brought, but not the reason why it was used.
```suggestion
```